packages/backend-rest-api/src/construct.ts

```diff
@@ -0,0 +1,261 @@
+import { Construct } from 'constructs';
+import { 
+  AuthorizationType, 
+  CfnAuthorizer, 
+  CfnMethod, 
+  CfnResource, 
+  CognitoUserPoolsAuthorizer, 
+  EndpointType, 
+  LambdaIntegration, 
+  RestApi 
+} from 'aws-cdk-lib/aws-apigateway';
+import { BackendOutputStorageStrategy } from '@aws-amplify/plugin-types';
+import { RestApiOutput } from '@aws-amplify/backend-output-schemas';
+import { CfnOutput, Stack } from 'aws-cdk-lib';
+import { AmplifyRestApiFactoryProps, HttpMethod, RouteConfig } from './types.js';
+import { AuthResources } from '@aws-amplify/plugin-types';
+
+/**
+ * Resources created by the REST API construct
+ */
+export type RestApiResources = {
+  /**
+   * The REST API instance
+   */
+  restApi: RestApi;
+
+  /**
+   * CloudFormation resources
+   */
+  cfnResources: {
+    /**
+     * The CloudFormation REST API resource
+     */
+    cfnRestApi: CfnResource;
+
+    /**
+     * The CloudFormation authorizer resource (if auth is configured)
+     */
+    cfnAuthorizer?: CfnAuthorizer;
+
+    /**
+     * The CloudFormation method resources
+     */
+    cfnMethods: Record<string, CfnMethod>;
+  };
+};
+
+/**
+ * Properties for the AmplifyRestApi construct
+ */
+export type AmplifyRestApiProps = AmplifyRestApiFactoryProps & {
+  /**
+   * Output storage strategy for storing REST API outputs
+   */
+  outputStorageStrategy: BackendOutputStorageStrategy<RestApiOutput>;
+
+  /**
+   * Auth resources for configuring authorizers (optional)
+   */
+  authResources?: AuthResources;
+};
+
+/**
+ * Construct for creating a REST API in an Amplify backend
+ */
+export class AmplifyRestApi extends Construct {
+  /**
+   * The resources created by this construct
+   */
+  readonly resources: RestApiResources;
+
+  /**
+   * Create a new AmplifyRestApi
+   */
+  constructor(scope: Construct, id: string, private readonly props: AmplifyRestApiProps) {
+    super(scope, id);
+
+    // Create the REST API
+    const restApi = new RestApi(this, 'RestApi', {
+      restApiName: props.name,
+      description: `REST API for ${props.name}`,
+      endpointTypes: [props.endpointType ?? EndpointType.REGIONAL],
+      defaultCorsPreflightOptions: props.cors === true ? {
+        allowOrigins: ['*'],
+        allowMethods: ['GET', 'POST', 'PUT', 'DELETE', 'PATCH', 'OPTIONS', 'HEAD'],
+        allowHeaders: ['Content-Type', 'X-Amz-Date', 'Authorization', 'X-Api-Key', 'X-Amz-Security-Token'],
+      } : props.cors || undefined,
+      defaultMethodOptions: props.defaultMethodOptions,
+      ...props.restApiProps,
+    });
+
+    // Create authorizer if auth resources are provided
+    let authorizer: CognitoUserPoolsAuthorizer | undefined;
+    if (props.authResources?.userPool) {
+      authorizer = new CognitoUserPoolsAuthorizer(this, 'CognitoAuthorizer', {
+        cognitoUserPools: [props.authResources.userPool],
+        authorizerName: `${props.name}-cognito-authorizer`,
+      });
+    }
```

Comment:
1. Make this user configurable, i.e. some paths are allowed for guests or unauthenticated, some paths are for authenticated and some might be controlled through cognito groups defined in cognito auth.

2. Add validations that the groups used here are configured in auth.